modules/standards/tag_governance_policy.tf

```diff
@@ -0,0 +1,11 @@
+data "template_file" "tag_governance" {
+  template = "${file("${path.module}/templates/resource_tag_policyRule.json.tmpl")}"
+}
+
+resource "azurerm_policy_definition" "tag" {
+  name         = "resourceTagGovernance"
+  policy_type  = "Custom"
+  mode         = "All"
+  display_name = "Mandatory 'managedBy' tag missing. E.g. 'managedBy' = '<Team or Individual Name>'"
```

Comment:
shouldn't we add this automatically via the jenkins-library first?
https://github.com/hmcts/cnp-jenkins-library/blob/master/vars/spinInfra.groovy#L49

Reply:
Good point. The answer is yes, we should. But this is just to create a policy structure and first policy. This is not yet assigned to any scope.

Reply:
Why is it not a good idea?

Reply:
We currently have 2 places from where tags are applied, Terraform and Jenkins. This creates confusion and sometimes overlapping situations. 2 places to manage tags. Tagging is a feature of terraform and if in future we will need to change/restructure the tags we will need to keep in mind.

Reply:
the jenkins-library invokes terraform, we can't store the tags in 100+ repos it's not maintainable

Reply:
You always store tags with modules that implement resource provisioning. Modules are created so that they can be used in multiple projects. If you use a module that doesn't create tags will always be a problem for any new projects or places where you will take that module.

Reply:
the issue here is not the modules, it's the consumers,
sure modules can add tags, but they need to know what to put there.

we can't break everywhere everytime we want to add a new tag,

controlling it centrally works far better

it did initially start out in the modules but it really doesn't work with a decentralised building blocks approach that we have

Reply:
why would you break to add tags? 
Tags are just a parameter that you add to the module and let the calling function decide what to send to that module with the help of variables and the corresponding values in tfvars/yaml file. You still control it centrally within a module of what tags can be passed to the module. It's the same case as passing the -var with terraform plan and apply or using a centralized tfvars file within terraform. Which gives you complete visibility of where/what/how your code is doing. You don't end up finding sam things implemented at different places/tools while troubleshooting.